src/Network/Network/NetworkManager/NetworkManagerSecurityAdminConfiguration/NetworkManagerSecurityAdminRuleCollection/NetworkManagerSecurityAdminRule/NetworkManagerSecurityAdminRuleBaseCmdlet.cs

```diff
@@ -37,16 +38,35 @@ public bool IsNetworkManagerSecurityAdminRulePresent(string resourceGroupName, s
             {
                 this.NetworkManagerSecurityAdminRuleOperationClient.Get(resourceGroupName, networkManagerName, configName, ruleCollectionName, name);
             }
-            catch (Microsoft.Rest.Azure.CloudException exception)
+            catch (CommonErrorResponseException commonException)
             {
-                if (exception.Response.StatusCode == HttpStatusCode.NotFound || exception.Response.StatusCode == HttpStatusCode.BadRequest)
+                if (commonException.Response.StatusCode == HttpStatusCode.NotFound || commonException.Response.StatusCode == HttpStatusCode.BadRequest)
                 {
                     // Resource is not present
                     return false;
                 }
 
                 throw;
             }
+            catch (Exception ex)
+            {
+                // Check if the unknown exception has a property named "StatusCode"
+                var statusCodeProperty = ex.GetType().GetProperty("StatusCode");
+
+                if (statusCodeProperty != null)
+                {
+                    var statusCode = (HttpStatusCode)statusCodeProperty.GetValue(ex);
```

Comment:
why do you need to pass ex to .GetValue()?

Reply:
to get the actual status code as it might differ at runtime and I wanted to make sure it matches the Http status code